aws_topology/stackstate_checks/aws_topology/resources/rds.py

```diff
@@ -1,89 +1,161 @@
-            instance_data = make_valid_data(instance_data_raw)
+    def process_one_instance(self, instance_arn):
+        instance_id = instance_arn.rsplit(":", 1)[-1]
+        for instance_data in self.collect_instances(DBInstanceIdentifier=instance_id):
             self.process_instance(instance_data)
 
-    def process_instance(self, instance_data):
-        instance_arn = instance_data['DBInstanceArn']
-        instance_id = instance_data['DBInstanceIdentifier']
-        tags = self.client.list_tags_for_resource(ResourceName=instance_arn).get('TagList') or []
-        instance_data['Tags'] = tags
-        instance_data['URN'] = [
-            "urn:endpoint:/%s" % instance_data['Endpoint']['Address']
-        ]
-        instance_data['Name'] = instance_id
-        instance_data.update(with_dimensions([{'key': 'DBInstanceIdentifier', 'value': instance_id}]))
-        self.emit_component(instance_arn, 'aws.rds_instance', instance_data)
-        vpc_id = instance_data['DBSubnetGroup']['VpcId']
-        self.emit_relation(instance_arn, vpc_id, 'uses service', {})
-        if instance_data.get('VpcSecurityGroups'):  # TODO agent.create_security_group_relations (but needs change?)
-            for security_group in instance_data['VpcSecurityGroups']:
-                self.emit_relation(instance_arn, security_group['VpcSecurityGroupId'], 'uses service', {})
-
-    def process_cluster(self, cluster_data):
-        cluster_id = cluster_data['DBClusterIdentifier']
-        cluster_arn = cluster_data['DBClusterArn']
-        cluster_data['Name'] = cluster_arn
-        cluster_data.update(with_dimensions([{'key': 'DBClusterIdentifier', 'value': cluster_id}]))
-        self.emit_component(cluster_arn, self.COMPONENT_TYPE, cluster_data)
-        for cluster_member in cluster_data['DBClusterMembers']:
-            db_identifier = cluster_member['DBInstanceIdentifier']
-            arn = self.agent.create_arn('AWS::RDS::DBInstance', self.location_info, db_identifier)
-            self.emit_relation(cluster_arn, arn, 'has_cluster_node', {})
-
-    EVENT_SOURCE = 'rds.amazonaws.com'
+    @transformation()
+    def process_instance(self, data):
+        instance = Instance(data.instance, strict=False)
+        instance.validate()
+        output = make_valid_data(data.instance)
+        instance_arn = instance.DBInstanceArn
+        instance_id = instance.DBInstanceIdentifier
+        output["Name"] = instance_id
+        output["Tags"] = data.tags
+        output["URN"] = ["urn:endpoint:/" + instance.Endpoint.Address]
+        output.update(with_dimensions([{"key": "DBInstanceIdentifier", "value": instance_id}]))
+        self.emit_component(instance_arn, "aws.rds_instance", output)
+        self.emit_relation(instance_arn, instance.DBSubnetGroup.VpcId, "uses service", {})
+        # TODO agent.create_security_group_relations (but needs change?)
+        for security_group in output.get("VpcSecurityGroups", []):
```

Comment:
Here the loop should not be ont the output. But rather on the schematic instance